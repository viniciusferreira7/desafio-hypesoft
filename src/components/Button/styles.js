import styled, { css } from 'styled-components';

export const Container = styled.button`
${({ theme, background, sizes }) => css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 1.2rem ${theme.spacings.large};
  letter-spacing: 0.02rem;
  text-transform: uppercase;
  background:${background ? theme.colors.secondaryColor : theme.colors.white};
  color:${background ? theme.colors.white : theme.colors.primaryColor};
  font-size: ${sizes}rem;
  border-radius: 5px;
  border: none;
  transition: all 500ms ease-in-out;
  cursor: pointer;

&:hover {
  background:${!background ? theme.colors.secondaryColor : theme.colors.white};
  color:${!background ? theme.colors.white : theme.colors.primaryColor};
}
`}
`;
