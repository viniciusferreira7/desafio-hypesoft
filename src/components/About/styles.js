import styled, { css } from 'styled-components';
import { Container as Button } from '../Button/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 2.8rem;
  grid-template-areas:'SrcImg' 'Description';
  padding-top: 10rem;
`}
`;

export const SrcImg = styled.div`
${({ theme, srcImg }) => css`
  grid-area: 'SrcImg';
  max-width: 56rem;
  width: 100%;
  height: 54.2rem;
  background-image: url(${srcImg});
  background-size: cover;
  border-radius: 1rem;
  box-shadow: 3.2rem 3.2rem   ${theme.colors.white};

  @media ${theme.media.lteMedium}{
    display: none;
  }
`}
`;

export const Description = styled.div`
${() => css`
  grid-area: 'Description';
  max-width: 67.2rem;

  > ${Button} {
    width: 16.2rem;
    height: 5.6rem;

    > img{
      margin-left: 1.1rem;
    }
  }
`}
`;

export const InfoHeading = styled.div`
${() => css`
`}
`;

export const InfoContainer = styled.div`
${({ theme }) => css`
  max-width: 67.2rem;
  display: flex;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    padding-bottom: 3.2rem;
  }
`}
`;

export const Group = styled.div`
${() => css`
  max-width: 20.6rem;
  display: flex;
  align-items: center;
  margin-right: 5rem;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;

export const GroupTwo = styled.div`
${() => css`
  width: 25.1rem;
  display: flex;
  align-items: center;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;
