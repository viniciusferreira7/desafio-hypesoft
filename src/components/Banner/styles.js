import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Button } from '../Button/styles';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  max-height:31.2rem;
  overflow: hidden;
  padding: 6.4rem 2%;
  position: relative;
  margin-top: 16.4rem;

  @media ${theme.media.lteMedium}{
      max-height:auto;
      padding-top: 3rem
    }

 >  ${GridDot}{
    opacity: 0.15;
    position: absolute;
    top: 6.4rem;
    left: 12.2rem;
  }


  ${Heading}{
      width:50rem;
      height: 9.6rem;
      margin: 0 auto;


      @media ${theme.media.lteMedium}{
        width: 100%;
        text-align: center;
        height: 15rem;
        margin-bottom: 7rem;
      }

      > span{
        color:#bc199c;
      }

      > span::before{
        display: none;
      }
    }

  > ${Button} {
      width: 17.7rem;
      height: 5.6rem;
      margin: 0 auto;
      margin-top: 7.2rem;

      > a{
        margin-right: 0.9rem;
      }
  }
`}
`;

export const SmallEllipse = styled.div`
  ${({ theme }) => css`
    width:7.6rem;
    height:7.6rem;
    border-radius: calc(7.6rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 2.6rem;
    right: 11.6rem;

    @media ${theme.media.lteMedium}{
      top: 8rem;
      right: 0;
    }
  `}
`;

export const HugeEllipse = styled.div`
  ${({ theme }) => css`
    width:52.9rem;
    height:52.9rem;
    border-radius: calc(52.9rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 13.9rem;
    right: 90.7rem;

    @media ${theme.media.lteMedium}{
      width:30rem;
      height:30rem;
      top: 0;
      right: 4rem;
    }
  `}
`;

export const ContainerGridDot = styled.div`
  ${() => css`
    position: absolute;
    top:15rem;
    right: 21rem;
    transform: rotate(-90deg);

    ${GridDot}{
      opacity: 0.3;

    }
  `}
`;
