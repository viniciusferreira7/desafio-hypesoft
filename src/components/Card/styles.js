import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.4), rgba(223, 227, 235, 0.4)), #FFFFFF;
  border-radius: 6px;
  width: 31.2rem;
  max-height: 35.1rem;
  transition: all 400ms ease-in-out;

  @media ${theme.media.lteMedium}{
    width: 18.5rem;
    padding-bottom: 2%;
    max-height: 40rem;

    > svg {
      width:2.5rem !important;
      height:2.5rem !important;
      margin: 0;
    }

    ${TextContainer}{
      text-align: center;
      font-size: 2rem;
      padding:2.2rem 0 3.2rem 0 !important ;

      > strong {
        margin-top: 1.6rem;
      }
    }

  }

    > svg{
      width:4rem;
      height:4rem;
      margin: 2.2rem 2.2rem 0 2.2rem;
      color: #242424;
    }

    ${TextContainer}{
      padding:2.2rem 0 3.2rem 2.8rem ;
      position: relative;

      > strong {
        width: 6.4rem;
        height: 0.3rem;
        background-color: #ED1CA6;
        margin-top: 1.6rem;
        margin-bottom: 2.8rem;
        position: absolute;
        bottom: -1.3rem;
        left:6rem ;
      }
    }

    > img{
      margin: 2.8rem 0 3.6rem 3.5rem;
    }


    &:hover{
      background: #FFFFFF;
      box-shadow: 0px 20px 48px rgba(10, 25, 49, 0.09);
      border-radius: 6px;
      border: 0.3rem solid #ED1CA6;

      ${TextContainer}{
        color:#ED1CA6 ;
      }

      > svg {
        color: #ED1CA6;
      }
    }

;
`}
`;
