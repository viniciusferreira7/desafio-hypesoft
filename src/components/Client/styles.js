import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  display: flex;
  align-items: center;
  width: 31.2rem;
  height: 10rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.4rem;
`}
`;

export const Photo = styled.div`
${({ theme, srcImg }) => css`
  width: 5.2rem;
  height: 5.2rem;
  border-radius: calc(5.2rem / 2);
  background-image:url(${srcImg});
  background-size: cover;
  background-color: ${theme.colors.gray_500};
  margin: 2.4rem;
  `}
`;

export const Info = styled.div`
${() => css`
  ${TextComponent}{
    text-transform: capitalize;
    padding: 0;
    padding-bottom: 0.6rem;

    > span{
      color: #ED1CA6
;
    }
  }
`}
`;
