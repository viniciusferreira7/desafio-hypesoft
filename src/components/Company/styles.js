import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding-top: 20rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    text-align: center;
  }

  ${Heading}{
    width: 312px;

    > strong{
      color:#bc199c;
      font-weight: 400;
    }
  }
`}
`;

export const ContainerLogo = styled.div`
${({ theme }) => css`
  display: flex;
  max-width: 93.6rem;
  max-height: 12.0rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    width: 100%;
    max-height: 96.6rem;
  }
`}
`;
export const Logo = styled.div`
${({ theme }) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 168px;
  height: 120px;
  margin-right: 2.4rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.8rem;
  transition: all 500ms ease-in-out;

  @media ${theme.media.lteMedium}{
    margin: 0 auto;
    margin-top: 3rem;
  }

  &:hover{
    border:1rem solid ${theme.colors.white};
  }

  >img{
    width: 11.304rem;
    height: 2.6rem;
  }

  >img:nth-of-type(5){
    width:2.756rem !important;
    height:1.205rem !important;
  }

;

`}
`;
