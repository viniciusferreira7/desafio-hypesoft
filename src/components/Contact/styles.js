import styled, { css } from 'styled-components';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding: 10rem;
  align-items: center;
  margin-top: 22.7rem;
  position: relative;

  ${GridDot}{
    position: absolute;
    top: 16%;
    right:12%;
    transform: rotate(-90deg);
    opacity: 0.3;

    @media ${theme.media.lteMedium}{
      display: none;
  }
  }

  @media ${theme.media.lteMedium}{
    padding: 2rem;
    flex-direction: column;
  }
`}`;
