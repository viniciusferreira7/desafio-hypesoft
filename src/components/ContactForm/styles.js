import styled, { css } from 'styled-components';
import { Container as Button } from '../Button/styles';

export const Container = styled.form`
${({ theme }) => css`
  width: 100%;
  max-width: 76rem;
  padding: 4.8rem;
  background: ${theme.colors.white};
  box-shadow: 0rem 1.2rem 2.4rem rgba(10, 25, 49, 0.02);
  border-radius: 0.8rem;

  @media ${theme.media.lteMedium}{
    padding: 2rem;
  }
  `}
`;
export const GroupInput = styled.form`
${({ theme }) => css`
  width: 100%;
  max-width: 66.4rem;
  min-height: 28.8rem;
  margin: 2.4rem 0 3.2rem 0;

  input{
    width: 100%;
    padding: 1.4rem 2rem;
    background: ${theme.colors.white};
    border: 0.1rem solid ${theme.colors.gray_100};
    color:${theme.colors.gray_500};
    border-radius: 0.4rem;
    outline: none;
    margin-bottom: 1.6rem;
    transition: all 300ms ease-in-out;

    &:focus{
      border: 0.2rem solid ${theme.colors.secondaryColor};
    }

    &:first-of-type{
      margin-right: 1.6rem;

      @media ${theme.media.lteMedium}{
        margin-right: 0;
      }
    }
  }

  textarea{
    padding: 1.4rem 2.4rem;
    max-width: 66.4rem;
    width: 100%;
    height: 15.2rem;
    resize: none;
    outline: none;
    border: 0.1rem solid ${theme.colors.gray_100};
    transition: all 300ms ease-in-out;

    &:focus{
      border: 0.2rem solid ${theme.colors.secondaryColor};
    }
  }

  ${Button}{
    margin-top: 3.2rem;
    max-width: 24rem;
    width: 100%;

    &:hover{
      border: 0.2rem solid ${theme.colors.secondaryColor};
    }

    &:active{
      background:${theme.colors.primaryColor};
      color:${theme.colors.white};
      border-color:${theme.colors.white};
    }

    >img{
      margin-left: 1.4rem;
    }
  }

  @media ${theme.media.lteMedium}{
  }
  `}
`;
export const TwoInputs = styled.form`
${({ theme }) => css`
  width: 100%;
  display: flex;
  justify-content: space-between !important;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
  }
  `}
`;
