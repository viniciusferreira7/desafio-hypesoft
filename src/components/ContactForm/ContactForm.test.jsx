import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import { ContactForm } from '.';

describe('<ContactForm/>', () => {
  it('should render with default values', () => {
    renderTheme(<ContactForm />);
    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Email/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Subjects/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/message/i)).toHaveStyle({
      'max-width': '66.4rem',
      width: '100%',
    });
  });
  it('should match snapshot', () => {
    const { container } = renderTheme(<ContactForm />);
    expect(container).toMatchSnapshot();
  });
});
