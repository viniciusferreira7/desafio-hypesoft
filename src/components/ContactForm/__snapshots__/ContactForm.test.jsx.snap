// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm/> should match snapshot 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1.2rem 3.2rem;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  text-transform: uppercase;
  background: #7D00FF;
  color: #ffff;
  font-size: 1.6rem;
  border-radius: 5px;
  border: none;
  -webkit-transition: all 500ms ease-in-out;
  transition: all 500ms ease-in-out;
  cursor: pointer;
  font-weight: 600;
}

.c5:hover {
  background: #ffff;
  color: #242424;
}

.c1 {
  color: #242424;
  font-weight: 500;
  font-size: 3.2rem;
}

.c1 > span {
  color: #7D00FF;
  position: relative;
  z-index: 2;
}

.c1 > span::before {
  content: '';
  opacity: 0.2;
  position: absolute;
  z-index: 1;
  top: 60%;
  left: 0;
  width: 90%;
  height: 3rem;
  background: #35016b;
}

.c0 {
  width: 100%;
  max-width: 76rem;
  padding: 4.8rem;
  background: #ffff;
  box-shadow: 0rem 1.2rem 2.4rem rgba(10,25,49,0.02);
  border-radius: 0.8rem;
}

.c2 {
  width: 100%;
  max-width: 66.4rem;
  min-height: 28.8rem;
  margin: 2.4rem 0 3.2rem 0;
}

.c2 input {
  width: 100%;
  padding: 1.4rem 2rem;
  background: #ffff;
  border: 0.1rem solid #DFE3EB;
  color: #A5A5A5;
  border-radius: 0.4rem;
  outline: none;
  margin-bottom: 1.6rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 input:focus {
  border: 0.2rem solid #7D00FF;
}

.c2 input:first-of-type {
  margin-right: 1.6rem;
}

.c2 textarea {
  padding: 1.4rem 2.4rem;
  max-width: 66.4rem;
  width: 100%;
  height: 15.2rem;
  resize: none;
  outline: none;
  border: 0.1rem solid #DFE3EB;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 textarea:focus {
  border: 0.2rem solid #7D00FF;
}

.c2 .c4 {
  margin-top: 3.2rem;
  max-width: 24rem;
  width: 100%;
}

.c2 .c4:hover {
  border: 0.2rem solid #7D00FF;
}

.c2 .c4:active {
  background: #242424;
  color: #ffff;
  border-color: #ffff;
}

.c2 .c4 >img {
  margin-left: 1.4rem;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify !important;
  -webkit-justify-content: space-between !important;
  -ms-flex-pack: justify !important;
  justify-content: space-between !important;
}

@media (max-width:768px) {
  .c0 {
    padding: 2rem;
    margin-top: 3rem;
  }
}

@media (max-width:768px) {
  .c2 input:first-of-type {
    margin-right: 0;
  }
}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <form
    class="c0"
  >
    <h1
      class="c1"
      color="#242424"
    >
      Let me know here.
    </h1>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <input
          id="name"
          name="name"
          placeholder="Full name"
          required=""
          type="text"
        />
        <input
          id="email"
          name="email"
          placeholder="Email address"
          required=""
          type="email"
        />
      </div>
      <input
        id="subject"
        name="subject"
        placeholder="Subjects"
        required=""
        type="subject"
      />
      <textarea
        cols="33"
        id="message"
        name="message"
        placeholder="Message"
        required=""
        rows="5"
      />
      <button
        class="c4 c5"
        sizes="1.6"
      >
        Send message 
        <img
          alt="send"
          src="assets/images/GroupPaper.png"
        />
      </button>
    </div>
  </form>
</div>
`;
